protocol/src/main/scala/org/alephium/protocol/model/Target.scala

```diff
@@ -58,9 +58,12 @@ object Target {
   }
 
   // scalastyle:off magic.number
-  def from(hashRate: HashRate, blockTime: Duration): Target = {
+  def from(hashRate: HashRate, blockTime: Duration)(implicit groupConfig: GroupConfig): Target = {
     val hashNeeded =
-      hashRate.value.multiply(BigInteger.valueOf(blockTime.millis)).divide(BigInteger.valueOf(1000))
+      hashRate.value
+        .multiply(BigInteger.valueOf(blockTime.millis))
+        // divide the hashrate due to: multiple chains and chain index encoding in block hash
+        .divide(BigInteger.valueOf((1000 * groupConfig.chainNum * groupConfig.chainNum).toLong))
```

Comment:
Question:
Why do we have `groupConfig.chainNum * groupConfig.chainNum` instead of `groupConfig.chainNum`?
I guess `chainNum` is already `groups * groups`, right?

Reply:
> chain index will use some hashrate
Guess that is where i got lost... :sweat_smile: